File: tmp_check_ics.py
import urllib.request
from datetime import datetime
import time

url = "https://ics.fixtur.es/v2/league/premier-league.ics"
req = urllib.request.Request(url, headers={'User-Agent': 'Mozilla/5.0'})
try:
    with urllib.request.urlopen(req) as resp:
        text = resp.read().decode('utf-8')
        lines = text.splitlines()
        
        matches = 0
        now = datetime.utcnow()
        for line in lines:
            if line.startswith("DTSTART"):
                val = line.split(":")[-1].replace("Z", "")
                try:
                    dt_obj = datetime.strptime(val, "%Y%m%dT%H%M%S")
                    if dt_obj.year == 2024 or dt_obj.year == 2025 or dt_obj.year == 2026:
                        delta = dt_obj - now
                        if -2 <= delta.days <= 14:
                            matches += 1
                except Exception as e:
                    pass
        print(f"Total lines: {len(lines)}. Found recent matches: {matches}")
except Exception as e:
    print("Error:", e)

File: tmp_check_ics.js
const https = require('https');

const checkUrl = (url, name) => {
    https.get(url, (res) => {
        let rawData = '';
        res.on('data', (chunk) => rawData += chunk);
        res.on('end', () => {
            const lines = rawData.split(/\r?\n/);
            let matches2026 = 0;
            let first2026 = null;
            let lastStarts = [];
            lines.forEach(line => {
                if(line.startsWith("DTSTART")) {
                    const start = line.split(":").pop();
                    if (start) lastStarts.push(start.trim());
                    if (line.includes("2026")) {
                        matches2026++;
                        if (!first2026) first2026 = line;
                    }
                }
            });
            console.log(`[${name}] Lines: ${lines.length}. 2026 matches: ${matches2026}`);
            if (matches2026 > 0) console.log(`[${name}] First 2026 match start: ${first2026}`);
            if (lastStarts.length > 0) {
                console.log(`[${name}] Last 3 matches in file:`, lastStarts.slice(-3));
            }
        });
    });
};

checkUrl('https://ics.fixtur.es/v2/league/premier-league.ics', 'EPL');
checkUrl('https://ics.fixtur.es/v2/league/russian-premier-league.ics', 'RPL');
